crates/services/p2p/src/service.rs

```diff
@@ -196,9 +197,20 @@ pub trait TaskP2PService: Send {
     ) -> anyhow::Result<()>;
 
     fn update_block_height(&mut self, height: BlockHeight) -> anyhow::Result<()>;
+
+    fn update_metrics<T>(&self, update_fn: T)
```

Comment:
This seems off to me. I don't feel like the `update_metrics` method should be on this trait. (And the method effectivly has nothing to do with metrics, it's more like `run_closure`)

It looks like you put it here because the `metrics: bool` is on `FuelP2PService`, but that is only used in one spot currently.

Metrics are handled globally. Maybe we consider handling capture of metrics globally as well? What do you think?

Reply:
I discussed this with Green today, and it looks like only 3/n modules use the metrics flag, and p2p being one of them. the others export metrics regardless of the flag being enabled or disabled (which imo is incorrect behaviour). 

I have a task on my to-do list to create an issue to make sure the metrics flag is propagated to all modules and exporting them only if it's enabled ~ perhaps this warrants a change or a trait from the metrics crate, but I'm holding off on that on account of premature optimization. 

happy to make it global if that makes it cleaner (which will need removal of the metrics field from P2pConfig)

Reply:
I'm fine with leaving this as is if we create a follow up issue that specifically includes removing the trait method and the metrics Boolean.  It sounds like we need to unify metrics architecture for all our services, in which case it could happen as part of the same issue/feature.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2144 :)

Reply:
> feature flag at compile time

maybe operators would want to turn metrics collection on/off using the cli argument, so I am in favour of keeping the flag with a sensible default :)

Reply:
Makes sense :+1: